config/v1/types_operatorhub.go

```diff
@@ -0,0 +1,84 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// OperatorHubKind is the string value of OperatorHub kind
+var OperatorHubKind = "OperatorHub"
+
+// OperatorHubSpec defines the desired state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubSpec struct {
+	// HubSources is the list of default OperatorSources and their configuration
+	HubSources []HubSource `json:"hubSources,omitempty"`
+}
+
+// OperatorHubStatus defines the observed state of OperatorHub
+// +k8s:openapi-gen=true
+type OperatorHubStatus struct {
+	// HubSourcesStatus encapsulates the result applying the configuration
+	HubSourcesStatus []HubSourceStatus `json:"hubSourcesStatus,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHub is the Schema for the operatorhubs API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +genclient:nonNamespaced
+type OperatorHub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OperatorHubSpec   `json:"spec,omitempty"`
+	Status OperatorHubStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OperatorHubList contains a list of OperatorHub
+type OperatorHubList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OperatorHub `json:"items"`
+}
+
+// HubSource is used to specify the OperatorSource and its configuration
+type HubSource struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+	// Disable is used to disable a default OperatorSource on cluster
+	Disable bool `json:"disable"`
+}
+
+// HubSourceStatus is used to reflect the current state of applying the
+// configuration to a default source
+type HubSourceStatus struct {
+	// Name is the name of one of the default OperatorSources
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
+	// Configuration is the state of the default OperatorSources configuration
+	Configuration map[string]string `json:"configuration,omitempty"`
```

Comment:
What is the key and what is the value for this?

Reply:
The key maps to each config element in `HubSource`. Today it will only be `disable`. The value will be "true" or "false". I kept it in this form to make it extensible.

Reply:
Why are we making extensible config?  We don't need that, we have a struct we can add fields to

Reply:
Generally you would never do this, you'd either make someone check the spec (and verify observedGeneration matches from status) or you'd just embed `ActiveSources` or something.

Reply:
Sure, I can make the change and embed `DisabledSources` that lists the disabled sources given that is what the configuration option is `disabled`. Does that work for you?